### PR TITLE
[ML] Deprecate X-Pack centric ML endpoints

### PR DESCRIPTION
**Pull request by droberts195**
This commit is part of our plan to deprecate and
ultimately remove the use of _xpack in the REST APIs.

Relates #35958

### CODE REVIEW

**Comment by droberts195** [comment]
The first commit on this PR contains the mechanical changes.

I fully expect that further more subtle changes will be required to get this fully working, but I wanted to get that first commit onto GitHub to reduce the risk of losing it.

File: x-pack/plugin/src/test/resources/rest-api-spec/api/ml.info.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "ml.info": {
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/ml.info.json]: Oh I see the file is deleted later

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by droberts195** [comment]
#36373 needs to be merged before the rolling upgrade tests will pass.

**Comment by droberts195** [comment]
run gradle build tests 2

**Comment by droberts195** [comment]
I opened elastic/kibana#26876 to adjust the UI to use the new endpoints.

Also @dolaru you may want to adjust the master branch of the QA framework to use `_ml` instead of `_xpack/ml`.

**Pull request merged**